tests/OrganizationMfaValidationTest.php

```diff
@@ -148,7 +148,202 @@ public function testAdminWithoutMfaCannotBecameMember()
+
+        try {
+            $this->client->getOrganization($this->organization['id']);
+            $this->fail('Admin having MFA disabled should not have access to the organization');
+        } catch (ClientException $e) {
+            $this->assertEquals('manage.mfaRequired', $e->getStringCode());
+        }
+
+        try {
+            $this->client->createProject($this->organization['id'], ['name' => 'Test']);
+            $this->fail('Admin having MFA disabled should not have access to the organization');
+        } catch (ClientException $e) {
+            $this->assertEquals('manage.mfaRequired', $e->getStringCode());
+        }
+
+        try {
+            $this->client->updateOrganization($this->organization['id'], []);
+            $this->fail('Admin having MFA disabled should not have access to the organization');
+        } catch (ClientException $e) {
+            $this->assertEquals('manage.mfaRequired', $e->getStringCode());
+        }
+
+        try {
+            $this->client->listOrganizationUsers($this->organization['id']);
+            $this->fail('Admin having MFA disabled should not have access to the organization');
+        } catch (ClientException $e) {
+            $this->assertEquals('manage.mfaRequired', $e->getStringCode());
+        }
+
+        try {
+            $this->client->listOrganizationInvitations($this->organization['id']);
+            $this->fail('Admin having MFA disabled should not have access to the organization');
+        } catch (ClientException $e) {
+            $this->assertEquals('manage.mfaRequired', $e->getStringCode());
+        }
+    }
+
+    public function testSuperAdminCanDeleteOrganizationIfMfaWasForced()
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUserWithMfa['email']]);
+
+        $this->normalUserWithMfaClient->enableOrganizationMfa($this->organization['id']);
+
+        $organization = $this->normalUserWithMfaClient->getOrganization($this->organization['id']);
+        $this->assertTrue($organization['mfaRequired']);
+
+        $organizationsListBeforeDelete = $this->client->listMaintainerOrganizations($this->testMaintainerId);
+
+        $this->client->deleteOrganization($this->organization['id']);
+
+        $organizationsListAfterDelete = $this->client->listMaintainerOrganizations($this->testMaintainerId);
+
+        $this->assertCount(count($organizationsListBeforeDelete) - 1, $organizationsListAfterDelete);
+    }
+
+    public function testSuperAdminCanListOrganizationProjectsIfMfaWasForced()
+    {
+        $this->client->addUserToOrganization($this->organization['id'], ['email' => $this->normalUserWithMfa['email']]);
+
+        $this->normalUserWithMfaClient->enableOrganizationMfa($this->organization['id']);
+
+        $organization = $this->normalUserWithMfaClient->getOrganization($this->organization['id']);
+        $this->assertTrue($organization['mfaRequired']);
```

Comment:
Tenhle assert tady vidím nějak často. To by spíš měl pokrejt test, který kontroluje zapnutí MFA a hotovo. Pak v každým testu kontroluješ to samý a zbytečně je to nepřehledný o co ti tam jde. `1 test by měl testovat 1 věc`

Chápu, že se jednou testuje orgranizace, projekt ..., ale to by měl být samostatný test, že se to zapne a pak to neřešit znova.

Reply:
fixed